monitor.py

```diff
@@ -293,49 +309,45 @@ def __init__(self, _db, collection=None, batch_size=None):
         self.model_class = cls.MODEL_CLASS
         super(SweepMonitor, self).__init__(_db, collection=collection)
 
-    def run(self):
+    def run_once(self, *ignore):
```

Comment:
Bu subclassing `run_once` (the "do something" method) instead of `run`, we get to reuse the code that manages the Timestamp and handles exceptions. That's why code to do those two things is removed from this method.